### PR TITLE
Request to Merge Manager

### PR DESCRIPTION
**Pull request by EBruchet**
**Fixes/Implementations:**
- Implemented `Tag` dropdown selection in the `CreateMenuItemDialog` and `UpdateMenuItemDialog` of the Manager view.
  - Note `onUpdate` in `App.js` has been updated to perform a `PATCH` request in place of a `PUT`, as the insertion/updating of the list of Tags would not work otherwise.
  - Also worth noting that while `UpdateMenuItemDialog` does work without visible issue, I had to use a very inelegant `setTimeout` and call `requestTags` twice in a row, because for some reason despite it updating the state of `menuItemTags`, the column of tags in the `MenuItemList` isn't updated unless called twice. Likely due to some asynchronous behaviour in JavaScript, but I wasn't able to use `Promises` to fix the issue - hence the **help wanted** tag.
- Deleted `style.css` as it was breaking styling on various pages.
- Deleted `select-task.css` and `select-table-num.css` and shifted their contents into `login.css` and `customer.css`, respectively (Please notify me if I made a mistake doing this, didn't notice anything but wouldn't be surprised if there were differences between Firefox/Chrome stylings again)
- Edited `login.css` as it also had view-breaking stylings for the other pages.

### CODE REVIEW

File: src/main/js/components/App.js [hunk 1]
```diff
@@ -357,12 +320,11 @@ export class App extends React.Component {
      */
     onUpdate(resource, updatedResource, resourceType) {
         client({
-            method: 'PUT',
+            method: 'PATCH',
```
Comment (Gabetn) [inline src/main/js/components/App.js:323]: Should be good, but will have to keep an eye on this due to the "_polymorphic_" nature of onUpdate, such that it won't affect the orders & diningsessions. Will test out modifying `Customer.js`   (_**handleTableNumberSelect**_) such that we can only pass the modified field we're interested in (e.g. tableAssignmentStatus) in the PATCH request as opposed to an entire modified entity. 
 Might come in handy for @adeeb27  & @dibbor implementation of their polymorphic modification function in `DiningSession.js`

File: src/main/js/components/App.js [hunk 2]
```diff
@@ -524,7 +486,7 @@ export class App extends React.Component {
                     <TransitionGroup>
                         <CSSTransition key={location.pathname} timeout={30000} classNames="fade" >
                             <Switch location={location}>
-                                <Route exact path={["/login", "/"]} component={Login}/>
+                                <Route exact path={"/login"} component={Login}/>
```
Comment (Gabetn) [inline src/main/js/components/App.js]: Why get rid of the **"/"** default route to login page? Shouldn't the app enter to login?

File: src/main/js/components/Customer.js [hunk 3]
```diff
@@ -2,20 +2,15 @@
 
 /** ----- NPM PACKAGE IMPORTS -----**/
 import React from "react";
-import "../../resources/static/css/style.css";
```
Comment (Gabetn) [inline src/main/js/components/Customer.js:5]: Deletions of these css files seems to have broken the styling of the `Customer` view, such that on **CustomerLanding** page there is now an extra vertical scroll bar. This occurs on both Firefox & Chrome.
image

**NOTE**, this seems to self-fix where after a timeout or perhaps scrolling it pops back into place and removes the extra scrollbar

File: src/main/js/components/Login.js [hunk 4]
```diff
@@ -56,10 +55,10 @@ export class Login extends React.Component {
...
         return (
             <div id="login-page" className={"page login-page"}>
-                <div className="loginbox">
-                    <div className="content">
-                        <form onSubmit={this.handleSubmit}>
-                            <h1>WELCOME</h1>
+                <div id="login-box">
+                    <div id="login-content">
+                        <form id="login-form" onSubmit={this.handleSubmit}>
+                            <h1 id="login-welcome-header">WELCOME</h1>
```
Comment (Gabetn) [inline src/main/js/components/Login.js:61]: Seems to have added added a verticalscrollbar to the loginpage (Low priority)

**Review by Gabetn — CHANGES_REQUESTED** [review]
### CSS Styling Issues
`Customer`:
1. Broken verticalScrollBar
2. Broken menuItems where there's no longer a **cart** button displayed when hovering over a menuitem

### General
Issues not so caused by this PR, but rather things I noticed we will need to work on during my review. 
1. **Synchronization**. Modifying menuitems in Manager doesn't reflect in customer
2. **TAGS** do we need to allow for modification of tags / deletion of tags?

### MenuItem.js
Typo: **Line 89** : "_this.props.menuItemLinks.prev.href_" the `/api/menuItems` doesn't have a "prev". Hence these warnings: 
**EDIT**: These are the result of clicking on a navigation button when the entire menu is loaded into the first page. 
image
**Continued**: I noticed the issue with _setTimeout_ and will investigate it further during our meeting today.

**Comment by EBruchet** [comment]
**CSS Styling Issues:**
`Customer:`
- Broken verticalScrollBar (**RESOLVED**)
- Broken menuItems where there's no longer a cart button displayed when hovering over a menuitem (**RESOLVED**)


**General**

> Issues not so caused by this PR, but rather things I noticed we will need to work on during my review.

- Synchronization. Modifying menuitems in Manager doesn't reflect in customer
  - Discussed in person, requires that we periodically execute `loadResourceFromServer()` in the Customer perspective to update what they see.
- TAGS do we need to allow for modification of tags / deletion of tags?
  - Not seen in any of the high level user stories, so @Gabetn and I have agreed to not make this priority as of now.

**Review by Gabetn — APPROVED** [review]

**Pull request merged**